### PR TITLE
Validation fixes

### PR DESCRIPTION
**Pull request by lavalamp**
Builds on #11; the last three commits are new.

### CODE REVIEW

File: schema/elements.go [hunk 1]
```diff
@@ -62,6 +63,21 @@ const (
 	Boolean = Scalar("boolean")
 )
 
+// ElementRelationship is an enum of the different possible relationships
+// between the elements of container types.
+type ElementRelationship string
```
Comment (apelisse) [inline schema/elements.go]: Topology? Didn't we spend time arguing about this word? :-)

Comment (lavalamp) [reply schema/elements.go]: Did we? Do you think "ElementTopology" is clearer to devs who haven't spent a bunch of time thinking about schemas?

Comment (apelisse) [reply schema/elements.go]: No forget it, it's fine like this.

**Pull request merged**